File: python/leetcode/find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        seen = {}
        for i in range(len(nums)):
            if nums[i] in seen:
                return nums[i]
            seen[nums[i]] = 1

File: java/leetcode/find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        HashMap<Integer, Boolean> seen = new HashMap();
        for (int i = 0; i < nums.length; i++) {
            if (seen.containsKey(nums[i])) { return nums[i]; }
            seen.put(nums[i], true);
        }
        return -1;
    }
}
